### PR TITLE
[Do Not Merge] hooks folder created, useLogin hook made from logic previously in Log…

### PR DESCRIPTION
**Pull request by leahkatherine**

### CODE REVIEW

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          }
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+      setLoading(false);
+    };
+    
+    const [code, setCode] = useState('');
+    const validateToken = async (code: 'string', email: string) => {
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:52]: Code: string 


Not code: 'string'

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            }
+          }
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+      setLoading(false);
+    };
+    
+    const [code, setCode] = useState('');
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:51]: Nit: move to top with other states

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return;
+      }
+  
+      await fetch('/api/auth-email', {
+        method: 'POST',
+        body: JSON.stringify({ email }),
+      })
+        .then((res) => {
+          if (res.status === 200) {
+            console.log(res);
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:35]: Nit: remove console log

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    return {
+      email,
+      setEmail,
+      auth,
+      loading,
+      handleLogin,
+      code,
+      setCode,
+      validateToken,
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:83]: Nit: Use Prettier to clean up code

File: libs/web/feature-auth/src/lib/pages/LoginPage.tsx [hunk 5]
```diff
@@ -2,6 +2,8 @@ import { PageLayout, validateEmail } from '@daruma-board/web/design-system';
 import { useState } from 'react';
 import { useRouter } from 'next/router';
 import { LoginBox } from '../components';
+
+
 export const LoginPage = () => {
   // TODO: move to hook
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/pages/LoginPage.tsx:8]: This file hasn't been updated, the logic needs to be replaced by useHook

**Review by aram-devdocs — CHANGES_REQUESTED** [review]

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx (hunk 1 above)
Comment (leahkatherine) [reply libs/web/feature-auth/src/lib/hooks/useLogin.tsx:52]: Not sure why this didn't update - my code has them removed from our convo we had yesterday?

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import { useRouter } from 'next/router';
+import { validateEmail } from '@daruma-board/web/design-system';
+
+export const useLogin = () => {
+  const [email, setEmail] = useState<string>('');
+  const [auth, setAuth] = useState<boolean>(false); // [auth, setAuth
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:7]: delete commented out

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  const [loading, setLoading] = useState<boolean>(false); // [loading, setLoading
+  const [code, setCode] = useState('');
+  const router = useRouter();
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:10]: nit: declare consts at top level ABOVE states

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      })
+      .catch((error) => {
+        console.error('Error:', error);
+      });
+    setLoading(false);
+  };
+
+  return {
+    email,
+    setEmail,
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:73]: do not return state change functions directly from hooks.

Create function such as:

```
const handleSetEmail = (email: string) => {
    setEmail(email)
    }
```

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    auth,
+    loading,
+    handleLogin,
+    code,
+    setCode,
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/hooks/useLogin.tsx:78]: same here

File: libs/web/feature-auth/src/lib/pages/LoginPage.tsx [hunk 10]
```diff
@@ -1,74 +1,12 @@
...
-          }
-        }
-      })
-      .catch((error) => {
-        console.error('Error:', error);
-      });
-    setLoading(false);
-  };
+export const LoginPage = () => {
+  const { email, setEmail, auth, code, setCode, handleLogin, validateToken, loading } = useLogin();
```
Comment (aram-devdocs) [inline libs/web/feature-auth/src/lib/pages/LoginPage.tsx]: make changes here on setEmail and setCode to handlers created in useLogin.tsx

**Review by aram-devdocs — CHANGES_REQUESTED** [review]
discussed during 1:1

File: libs/web/feature-auth/src/lib/hooks/useLogin.tsx (hunk 3 above)
Comment (leahkatherine) [reply libs/web/feature-auth/src/lib/hooks/useLogin.tsx]: Sorry just saw this before i made the most recent pushed. I will re commit

**Review by aram-devdocs — APPROVED** [review]

**Pull request merged**